Sources/JWTKit/ECDSA/ECDSAKey.swift

```diff
@@ -59,8 +59,45 @@ public final class ECDSAKey: OpenSSLKey {
     init(_ c: OpaquePointer) {
         self.c = c
     }
+    
+    convenience init(x: String, y: String, curve: Curve = .p521) throws {
+        guard let c = CJWTKitBoringSSL_EC_KEY_new_by_curve_name(curve.cName) else {
+            throw JWTError.signingAlgorithmFailure(ECDSAError.newKeyByCurveFailure)
+        }
+        
+        guard let bnX = BN.convert(x) else {
+            throw JWTError.generic(identifier: "ecCoordinates", reason: "Unable to interpret x as BN");
+        }
+        guard let bnY = BN.convert(y) else {
+            throw JWTError.generic(identifier: "ecCoordinates", reason: "Unable to interpret y as BN");
+        }
+
+        if (1 != CJWTKitBoringSSL_EC_KEY_set_public_key_affine_coordinates(c, bnX.c, bnY.c)) {
+            throw JWTError.generic(identifier: "ecCoordinates", reason: "Unable to set public key");
+        }
+
+        self.init(c)
+    }
 
     deinit {
         CJWTKitBoringSSL_EC_KEY_free(self.c)
     }
+
+    public func getParameters() throws -> Parameters {
```

Comment:
Since this only throws one error instance anyway, I think a computed property and optional is fine here. 

```swift
var parameters: Parameters? { get }
```